boa/environment.py

```diff
@@ -152,10 +170,15 @@ def reset_gas_used(self):
     @contextlib.contextmanager
     def anchor(self):
         snapshot_id = self.evm.snapshot()
+        # Remember new contracts from this snapshot
+        self._new_contracts = []
```

Comment:
i don't think this is correct in the presence of nested anchors, since the outer journal will get discarded

Reply:
I think we need an actual journal to encapsulate any data that changes in between snapshots

Reply:
or maybe something really "clever", like a weakdict